src/SQLiteAsync.cs

```diff
@@ -50,74 +51,247 @@ public interface ISQLiteAsyncConnection
+#if NET8_0_OR_GREATER
+			[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.All)]
+#endif
+			T,
+#if NET8_0_OR_GREATER
+			[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.All)]
+#endif
+			T2> (CreateFlags createFlags = CreateFlags.None)
 			where T : new()
 			where T2 : new();
-		Task<CreateTablesResult> CreateTablesAsync<T, T2, T3> (CreateFlags createFlags = CreateFlags.None)
+		Task<CreateTablesResult> CreateTablesAsync<
+#if NET8_0_OR_GREATER
+			[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.All)]
+#endif
+			T,
+#if NET8_0_OR_GREATER
+			[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.All)]
+#endif
+			T2,
+#if NET8_0_OR_GREATER
+			[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.All)]
+#endif
+			T3> (CreateFlags createFlags = CreateFlags.None)
 			where T : new()
 			where T2 : new()
 			where T3 : new();
-		Task<CreateTablesResult> CreateTablesAsync<T, T2, T3, T4> (CreateFlags createFlags = CreateFlags.None)
+		Task<CreateTablesResult> CreateTablesAsync<
+#if NET8_0_OR_GREATER
+			[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.All)]
+#endif
+			T,
+#if NET8_0_OR_GREATER
+			[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.All)]
+#endif
+			T2,
+#if NET8_0_OR_GREATER
+			[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.All)]
+#endif
+			T3,
+#if NET8_0_OR_GREATER
+			[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.All)]
+#endif
+			T4> (CreateFlags createFlags = CreateFlags.None)
 			where T : new()
 			where T2 : new()
 			where T3 : new()
 			where T4 : new();
-		Task<CreateTablesResult> CreateTablesAsync<T, T2, T3, T4, T5> (CreateFlags createFlags = CreateFlags.None)
+		Task<CreateTablesResult> CreateTablesAsync<
+#if NET8_0_OR_GREATER
+			[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.All)]
+#endif
+			T,
+#if NET8_0_OR_GREATER
+			[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.All)]
+#endif
+			T2,
+#if NET8_0_OR_GREATER
+			[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.All)]
+#endif
+			T3,
+#if NET8_0_OR_GREATER
+			[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.All)]
+#endif
+			T4,
+#if NET8_0_OR_GREATER
+			[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.All)]
+#endif
+			T5> (CreateFlags createFlags = CreateFlags.None)
 			where T : new()
 			where T2 : new()
 			where T3 : new()
 			where T4 : new()
 			where T5 : new();
+#if NET8_0_OR_GREATER
+		[RequiresUnreferencedCode ("This method requires 'DynamicallyAccessedMemberTypes.All' on each input 'Type' instance.")]
+#endif
 		Task<CreateTablesResult> CreateTablesAsync (CreateFlags createFlags = CreateFlags.None, params Type[] types);
-		Task<IEnumerable<T>> DeferredQueryAsync<T> (string query, params object[] args) where T : new();
+		Task<IEnumerable<T>> DeferredQueryAsync<
+#if NET8_0_OR_GREATER
+			[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.All)]
```

Comment:
Do we really need all? The API really only hits the public properties and public ctors. That said, after reading the docs, the whole type will get included either way. But just a note that in the future I might narrow these down.

Reply:
We should be able to reduce this if we rework some internal APIs to better retrieve the properties. I can do that in a follow up. I wanted to take things gradually here and just do a first pass in this PR without changing code 🙂

In theory we might be able to scope this down to just `PublicProperties`, or that and `PublicFields`.

Reply:
@Sergio0694 Do you know whether the compiler will pick up attributes inherited from `DynamicallyAccessedMembers`? I was thinking a gross, but effective way to get rid of all the conditional compiles would be to introduce my own attribute that I can inherit from `DynamicallyAccessedMembers` for .net8+ while just having a do-nothing shim for netstandard.

Reply:
That's not supported. However, you could add (shameless plug) [PolySharp](https://github.com/Sergio0694/PolySharp) as a private dependency, have it only generate the `[DynamicallyAccessedMembers]` polyfill, and then just use it in all targets. The main reason why I haven't done that here is that I see you also support an "embedded" (as in, source only) mode for SQLite.net, and having polyfill types mixed with embedded mode can cause problems in some scenarios due to conflicts 😅